brave-core/src/main/java/com/github/kristofa/brave/InheritableServerClientAndLocalSpanState.java

```diff
@@ -0,0 +1,123 @@
+package com.github.kristofa.brave;
+
+import java.util.Deque;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import com.github.kristofa.brave.internal.Util;
+import com.twitter.zipkin.gen.Endpoint;
+import com.twitter.zipkin.gen.Span;
+
+/**
+ * {@link ServerClientAndLocalSpanState} implementation that keeps trace state using {@link InheritableThreadLocal}
+ * variables and a {@link Deque} of local spans.
+ */
+public final class InheritableServerClientAndLocalSpanState implements ServerClientAndLocalSpanState {
+
+    private final InheritableThreadLocal<ServerSpan> currentServerSpan =
+            new InheritableThreadLocal<ServerSpan>() {
+                @Override
+                protected ServerSpan initialValue() {
+                    return ServerSpan.EMPTY;
+                }
+            };
+
+    private final InheritableThreadLocal<Span> currentClientSpan = new InheritableThreadLocal<>();
```

Comment:
The `<>` here isn't compatible with the language-level of this module (Java 6) on the current `master` branch